website/docs/help-and-support/troubleshooting-guide/widget-errors.md

```diff
@@ -1,38 +1,66 @@
+### Source data exceeds 50 fields
 
 You see an error message **Source data exceeds 50 fields. Please update the source data** when you try to bind the query/API response to the [source data ](../../reference/widgets/json-form#source-data)property.
 
-**Error message:**
+#### Error message
 
 Source data exceeds 50 fields. Please update the source data.
 
 ![When the data had more than 50 fields](</img/Troubleshooting__Widget_Errors__JSON_Form_Errors__Source_Exceeds_50_Fields.png>)
 
-### Cause
+#### Cause
 
 The problem can be caused when you are trying to bind:
 
 * A large array of multiple JSON objects
 * A huge JSON object which has a lot of fields
 * The whole query data rather than a selected row or triggered row in a table
 
-### Solution
+#### Solution
 
 To determine if the problem is caused due to:
 
-* **A large array or a huge JSON object** - You can relook at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
-* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either selected row / triggered row to bind.
+* **A large array or a huge JSON object** - You can re-look at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
+* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either the selected row / triggered row to bind.
 
 Once you have figured out the new structure for the data, head to the [source data](../../reference/widgets/json-form#source-data) field to make changes.
 
+## Query/API/JS object execution errors
+You may see below errors when execute an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) using a widget property.
+
+### Action can't be triggered
+You may see the error when binding an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) execution to a widget property that expects data, and can't be used to trigger an action. 
+
+#### Error message
+Action cannot be triggered from this field.
+
+**To-Do to add screenshot**
+
+#### Cause
+An **action** is referred to as an operation that could be performed by executing an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/). 
+
+:::info
+Action can only be performed by binding it to an [async](../../core-concepts/writing-code/workflows#async-fields) field.
```

Comment:
I think this :::info might work better when combined with the next sentence ("When you bind an action...") since they are closely related. Maybe by either putting them both in the :::info block, or just having them both be plain text.

Reply:
True, removed the info.